src/stream/src/executor/backfill/arrangement_backfill.rs

```diff
@@ -450,54 +451,58 @@ where
                     "backfill_finished_wait_for_barrier"
                 );
                 // If not finished then we need to update state, otherwise no need.
-                if let Message::Barrier(barrier) = &msg
-                    && !is_completely_finished
```

Comment:
Previously we wait for this flag to pass. However `is_completely_finished` could be true on recovery, if backfill is completed already.

In that case, we should just yield the barrier and stop waiting.

Reply:
IIUC, after recovery, the backfill state may be inconsistent with the `CreateMaterializedTracker` state, so we need to notify the progress to finish every time after recovery.

Reply:
Yes, we will always notify see https://github.com/risingwavelabs/risingwave/pull/14563/files#r1464421419. It is outside of the if-else block.